monitor/dbt_project/macros/get_tests_sample_data.sql

```diff
@@ -40,12 +41,40 @@
             {% do elementary_tests_allowlist_status.append('pass') %}
         {% endif %}
         {%- if (test_type == 'dbt_test' and status in ['fail', 'warn']) or (test_type != 'dbt_test' and status in elementary_tests_allowlist_status) -%}
-            {% set test_rows_sample = elementary_internal.get_test_rows_sample(test_results_query, test_type, metrics_sample_limit) %}
+            {# Dimension anomalies return multiple dimensions for the test rows sample, and needs to be handle differently. #}
+            {# Currently we show only the anomalous for all of the dimensions. #}
+            {% if test_sub_type == 'dimension' %}
+                {% set dimension_test_result_query %}
+                    select *
+                    from ({{test_results_query}})
+                    where is_anomalous
+                {% endset %}
+                {% set test_rows_sample = elementary_internal.get_test_rows_sample(dimension_test_result_query, test_type, metrics_sample_limit) %}
+                {% set anomalous = [] %}
+                {% for sample in test_rows_sample %}
+                    {% set anomalous_sample = {
+                        'end_time': sample['end_time'],
```

Comment:
Currently the UI handles converting `end_time` to the right title.
This is important because our code breaks if it is not getting an `end_time`.
Because this is a temporary solution, I prefer to maintain this solution in one place (instead of handling it in multiple files) and that the UI will adjust the naming here.